presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -113,4 +122,31 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
                 .setNullsFraction((nullsCountLeft + nullsCountRight) / newRowCount)
                 .build();
     }
+
+    public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    {
+        checkArgument(new HashSet<>(left.getSymbolsWithKnownStatistics()).equals(new HashSet<>(right.getSymbolsWithKnownStatistics())));
+
+        PlanNodeStatsEstimate.Builder statsBuilder = PlanNodeStatsEstimate.builder();
+
+        for (Symbol symbol : left.getSymbolsWithKnownStatistics()) {
+            SymbolStatsEstimate leftSymbolStats = left.getSymbolStatistics(symbol);
+            SymbolStatsEstimate rightSymbolStats = right.getSymbolStatistics(symbol);
+            StatisticRange leftRange = StatisticRange.from(leftSymbolStats);
+            StatisticRange rightRange = StatisticRange.from(rightSymbolStats);
+            StatisticRange intersection = leftRange.intersect(rightRange);
+
+            statsBuilder.addSymbolStatistics(
+                    symbol,
+                    SymbolStatsEstimate.builder()
+                            .setStatisticsRange(intersection)
+                            // it does matter how many nulls are preserved, the intersting point is the fact if there are nulls or not
```

Comment:
// it does matter how many nulls are preserved, the intersting point is the fact if there are nulls **ON BOTH SIDES** or not